test/unit/strainer_unit_test.rb

```diff
@@ -145,4 +145,13 @@ def test_global_filter_clears_cache
     Strainer.global_filter(LateAddedFilter)
     assert_equal 'filtered', Strainer.create(nil).invoke('late_added_filter', 'input')
   end
+
+  def test_add_filter_returns_nil_when_a_filter_module_was_added
+    a = Module.new
+    strainer = Context.new.strainer
+    result1 = strainer.class.add_filter(a)
+    result2 = strainer.class.add_filter(a)
```

Comment:
I don't think you need to open another PR.  You could just change the test in this PR.